In [1]:
import pandas as pd

everest = pd.read_csv("everest.csv")


/var/folders/j7/bs_78v014_z3vp4jh1n05bth0000gn/T/ipykernel_13307/2508954031.py:3: DtypeWarning: Columns (80,81,82) have mixed types. Specify dtype option on import or set low_memory=False.
  everest = pd.read_csv("everest.csv")


In [2]:
# Converting 'deathdate' column into datetime format

everest['deathdate'] = pd.to_datetime(everest['deathdate'], errors='coerce')

# Finding the first person who died in the Mt. Everest
first_death = everest.loc[
    everest[everest['death']]['deathdate'].idxmin(),
    ['fname', 'lname', 'sex','citizen', 'calcage', 'deathhgtm', 'deathdate', 'deathnote']
]
print(first_death)


fname                                       Alexander Mitchell
lname                                                   Kellas
sex                                                          M
citizen                                                     UK
calcage                                                     52
deathhgtm                                                 5700
deathdate                                  1921-06-05 00:00:00
deathnote    Died of illness on approach march (heart failu...
Name: 5691, dtype: object


/var/folders/j7/bs_78v014_z3vp4jh1n05bth0000gn/T/ipykernel_13307/4222574070.py:3: UserWarning: Could not infer format, so each element will be parsed individually, falling back to `dateutil`. To ensure parsing is consistent and as-expected, please specify a format.
  everest['deathdate'] = pd.to_datetime(everest['deathdate'], errors='coerce')


In [3]:
fname, lname, citizen, calcage, deathhgtm, deathnote, deathdate = (
    first_death['fname'],
    first_death['lname'],
    first_death['citizen'],
    first_death['calcage'],
    first_death['deathhgtm'],
    first_death['deathnote'],
    first_death['deathdate'].strftime('%B %d, %Y'),   
)

print(
    f"The first person who died during the expedition on Mt. Everest was {fname} {lname} from the {citizen} on {deathdate}. " 
f"He was {calcage} years old, and he died at an altitude of {deathhgtm} meters. "
f"{lname} {deathnote[0].lower() + deathnote[1:]}."
)

The first person who died during the expedition on Mt. Everest was Alexander Mitchell Kellas from the UK on June 05, 1921. He was 52 years old, and he died at an altitude of 5700 meters. Kellas died of illness on approach march (heart failure) near pass to Kampa Dzong.


In [89]:
# Finding the last person who died on Mt. Everest based on data up to the Autumn–Winter 2024 expedition
last_death = everest.loc[
    everest[everest['death']]['deathdate'].idxmax(),
    ['fname', 'lname', 'sex','citizen', 'calcage', 'deathhgtm', 'deathdate', 'deathtype', 'deathclass','deathnote']
]
print(last_death)

fname            Joshua Cheruiyot
lname                       Kirui
sex                             M
citizen                     Kenya
calcage                        40
deathhgtm                    8800
deathdate     2024-05-22 00:00:00
deathtype                       1
deathclass                      4
deathnote                     NaN
Name: 24465, dtype: object


In [91]:
fname, lname, citizen, calcage, deathhgtm, deathdate = (
    last_death['fname'],
    last_death['lname'],
    last_death['citizen'],
    last_death['calcage'],
    last_death['deathhgtm'],
    last_death['deathdate'].strftime('%B %d, %Y'),   
)

print(
    f"The last person who died during the expedition on Mt. Everest, based on data through the Autumn–Winter 2024 season," 
    f" was {fname} {lname} from {citizen} on {deathdate}."
    f" He was {calcage} years old, and he died at an altitude of {deathhgtm} meters."
)

The last person who died during the expedition on Mt. Everest, based on data through the Autumn–Winter 2024 season, was Joshua Cheruiyot Kirui from Kenya on May 22, 2024. He was 40 years old, and he died at an altitude of 8800 meters.


In [39]:
# Finding the oldest person who died on Mt. Everest based on data up to the Autumn–Winter 2024 expedition

old_death = everest.loc[
    everest[everest['death']]['calcage'].idxmax(),
    ['fname', 'lname', 'sex','citizen', 'calcage', 'deathhgtm', 'deathdate', 'deathnote']
]

old_death

fname                          Min Bahadur
lname                             Sherchan
sex                                      M
citizen                              Nepal
calcage                                 85
deathhgtm                             5350
deathdate              2017-05-06 00:00:00
deathnote    Died of apparent heart attack
Name: 19205, dtype: object

In [50]:
fname, lname, citizen, calcage, deathhgtm, deathnote, deathdate = (
    old_death['fname'],
    old_death['lname'],
    old_death['citizen'],
    old_death['calcage'],
    old_death['deathhgtm'],
    old_death['deathnote'],
    last_death['deathdate'].strftime('%B %d, %Y'),   
)

print(
    f"The oldest person who died during the expedition on Mt. Everest, based on data through the Autumn–Winter 2024 season," 
    f" was {fname} {lname} from {citizen} on {deathdate}."
    f" He was {calcage} years old, and he died at an altitude of {deathhgtm} meters."
    f" According to the BBC story(https://www.bbc.com/news/world-asia-39829636), he was a former British Gurkha soldier who was" 
    f"attempting to reclaim the record from Japan's Yuichiro Miura, who had climbed Everest at age 80 in 2013. Miura had previously" 
    f"broken Sherchan's own record, which was set when he was 76 years old in 2008."

)

The oldest person who died during the expedition on Mt. Everest, based on data through the Autumn–Winter 2024 season, was Min Bahadur Sherchan from Nepal on May 22, 2024. He was 85 years old, and he died at an altitude of 5350 meters. According to the BBC story(https://www.bbc.com/news/world-asia-39829636), he was a former British Gurkha soldier who wasattempting to reclaim the record from Japan's Yuichiro Miura, who had climbed Everest at age 80 in 2013. Miura had previouslybroken Sherchan's own record, which was set when he was 76 years old in 2008.


In [45]:
# Finding the youngest person who died in the Mt. Everest 

young_death = everest.loc[
   everest[(everest['death'] == 1) & (everest['calcage'] > 0) & (everest['calcage'].notna())]['calcage'].idxmin(),
   ['fname', 'lname', 'sex','citizen', 'calcage', 'deathhgtm', 'deathdate']
]

print(young_death)


fname                   Chhewang
lname                     Sherpa
sex                            M
citizen                    Nepal
calcage                       19
deathhgtm                      0
deathdate    2014-05-01 00:00:00
Name: 16878, dtype: object


In [47]:
# Are there multiple people who died at the youngest age?
youngest_age = young_death['calcage']
multiple_youngest = everest[
   everest['death'] & 
   (everest['calcage'] == youngest_age)
][['fname', 'lname', 'sex', 'citizen', 'calcage', 'status', 'deathhgtm', 'deathdate','deathtype','residence', ]]

print(f"\nAll people who died at age {youngest_age}:")

print(multiple_youngest)

# Note: the deathtype column
# 7  – Avalanche
# 11 – Other



All people who died at age 19:
          fname   lname sex citizen  calcage             status  deathhgtm  \
16878  Chhewang  Sherpa   M   Nepal       19  H-A Worker (Cook)          0   
17076     Pemba  Sherpa   M   Nepal       19            BC Cook       5350   

       deathdate  deathtype               residence  
16878 2014-05-01         11    Nunthala, Solukhumbu  
17076 2015-04-25          7  Taksindu-5, Solukhumbu  


In [52]:
print(
    f"One of the youngest people who died on Mt. Everest was Pemba Sherpa from the Solukhumbu district, the district where Everest" 
    f"is located. He died on April 25, 2015, due to an avalanche (deathtype #7) caused by the earthquake in Nepal." 
    f"He was 19 years old and was at Base Camp working as a Sherpa cook."
)

One of the youngest people who died on Mt. Everest was Pemba Sherpa from the Solukhumbu district, the district where Everestis located. He died on April 25, 2015, due to an avalanche (deathtype #7) caused by the earthquake in Nepal.He was 19 years old and was at Base Camp working as a Sherpa cook.


In [6]:
print(f"Total deaths recorded on Mt. Everest from the very first on {first_death['deathdate'].strftime('%B %d, %Y')},"
      f" to the latest on {last_death['deathdate'].strftime('%B %d, %Y')}, is {everest['death'].sum()}.")

Total deaths recorded on Mt. Everest from the very first on June 05, 1921, to the latest on May 22, 2024, is 353.


## Total deaths recorded on Mt. Everest from the very first on June 05, 1921, to the latest on May 22, 2024, is 353.

In [7]:
# Everest deaths per year

everest_death = (
    everest.groupby('myear')['death']
    .sum()
    .loc[lambda x: x > 0]
    .sort_values(ascending=False)
)

# Converting Series to DataFrame
everest_death_df = everest_death.reset_index().rename(columns={'myear': 'year', 'death': 'total_death'})

# Saving file
everest_death_df.to_csv("everest_total_death_by_year.csv", index=False)


In [8]:
# Mapping dictionary for death type codes

# Death type (cause) codes in 'deathtype' column:

# 0  – Unspecified
# 1  – AMS (Acute Mountain Sickness)
# 2  – Exhaustion
# 3  – Exposure / Frostbite
# 4  – Fall
# 5  – Crevasse
# 6  – Icefall collapse
# 7  – Avalanche
# 8  – Falling rock/ice
# 9  – Disappearance (unexplained)
# 10 – Illness (non-AMS)
# 11 – Other
# 12 – Unknown


deathtype_map = {
    0: "Other/Unknown",
    1: "Acute Mountain Sickness",
    2: "Exhaustion",
    3: "Exposure/Frostbite",
    4: "Fall",
    5: "Crevasse",
    6: "Icefall collapse",
    7: "Avalanche",
    8: "Falling rock/ice",
    9: "Disappearance",
    10: "Illness(non-AMS)",
    11: "Other/Unknown",
    12: "Other/Unknown"
}

# The 'deathtype' column contains numerical codes representing causes of death, which are taken directly from the Himalayan Database Guide, 
# available at: https://www.himalayandatabase.com/downloads/Himalayan%20Database%20Guide.pdf


In [9]:
# Mapping dictionary for death classification codes

# 0 = Unknown or unspecified classification
# 1 = Death en route to or from base camp
# 2 = Death at base camp
# 3 = Death during the route preparation phase before a summit bid commences
# 4 = Death during the ascent phase of a summit bid whether successful or not
# 5 = Death during the descent phase of a summit bid
# 6 = Death during the route evacuation phase after all summit bids are completed
# 7 = Other or unknown classification of death

# Mapping dictionary for death classification codes

deathclass_map = {
    0: "Unknown",
    1: "En route to/from base camp",
    2: "At base camp", 
    3: "During route preparation",
    4: "Ascent phase",
    5: "Descent phase",
    6: "Route evacuation",
    7: "Other"
}

# The 'deathclass' column contains numerical codes representing phase of death, which are taken directly from the Himalayan Database Guide, 
# available at: https://www.himalayandatabase.com/downloads/Himalayan%20Database%20Guide.pdf


In [10]:
# Filtering deaths 
death_data = everest[everest['death'] == True]

# Mapping classification 
death_data['classification'] = death_data['deathclass'].map(deathclass_map)

# Mapping cause of death 
death_data['cause'] = death_data['deathtype'].map(deathtype_map)

# Adding a 'group' column to the dataset; values will be 'Sherpa' and 'Climber':
death_data['group'] = death_data['sherpa'].apply(lambda x: 'Sherpa' if x else 'Climber')

# Grouping Sherpas vs. Climbers by their causes of death

cause_of_death = (
    death_data
    .groupby(['group', 'cause'])
    .size()
    .reset_index(name='count')
    .sort_values(by=['group', 'count'], ascending=[True, False])
)

# Grouping Sherpas vs. Climbers by the phase of climbing during which they died

phase_of_death = (
    death_data
    .groupby(['group', 'classification'])
    .size()
    .reset_index(name='count')
    .sort_values(by=['group', 'count'], ascending=[True, False])
)

# Save these combined datasets for visualization
cause_of_death.to_csv("cause_of_death.csv", index=False)
phase_of_death.to_csv("phase_of_death.csv", index=False)


/var/folders/j7/bs_78v014_z3vp4jh1n05bth0000gn/T/ipykernel_13307/3429928227.py:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  death_data['classification'] = death_data['deathclass'].map(deathclass_map)
/var/folders/j7/bs_78v014_z3vp4jh1n05bth0000gn/T/ipykernel_13307/3429928227.py:8: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  death_data['cause'] = death_data['deathtype'].map(deathtype_map)
/var/folders/j7/bs_78v014_z3vp4jh1n05bth0000gn/T/ipykernel_13307/3429928227.py:11: SettingWithCopyWarning: 
A value

In [11]:
# Pivot the cause data to wide format
cause_of_death_wide = (
    cause_of_death
    .pivot(index='cause', columns='group', values='count')
    .fillna(0)
    .astype(int)
    .reset_index()
)

# Save wide-format cause data
cause_of_death_wide.to_csv("cause_of_death_wide.csv", index=False)

# Pivot the classification data to wide format
phase_of_death_wide = (
    phase_of_death
    .pivot(index='classification', columns='group', values='count')
    .fillna(0)
    .astype(int)
    .reset_index()
)

# Save wide-format classification data
phase_of_death_wide.to_csv("phase_of_death_wide.csv", index=False)


In [12]:

# Country-wise deaths on Mt. Everest

everest_death_country = everest[everest['death'] == True]['citizen'].value_counts()

# Adding 'Other' column for countries with less than 5 deaths and removing those countries

if (everest_death_country < 5).any():
    everest_death_country['Other'] = everest_death_country[everest_death_country < 5].sum()

# Keeping only countries with >= 5 deaths and 'Other'
everest_death_country = everest_death_country[everest_death_country >= 5]

everest_death_country.to_csv('everest_death_country.csv')

everest_death_country

citizen
Nepal             144
India              29
UK                 22
Japan              19
USA                17
China              12
S Korea            11
Russia              9
Australia           9
Germany             7
Poland              7
France              6
Canada              5
Czechoslovakia      5
Spain               5
Other              46
Name: count, dtype: int64

In [13]:

# Counting of deaths grouped by 'sherpa' and 'sex'

death_counts = (
    everest[everest['death'] == True]
    .groupby(['sherpa', 'sex'])
    .size()
    .reset_index(name='total_death')
)

# Replacing True/False values with 'Sherpa' and 'Climber'

death_counts['group'] = death_counts['sherpa'].map({True: 'Sherpa', False: 'Climber'})

# Keep only relevant columns
death_counts = death_counts[['group', 'sex', 'total_death']]

death_counts.to_csv("death_sherpa_vs_climbers.csv", index=False)

death_counts

group sex  total_death
0  Climber   F           13
1  Climber   M          224
2   Sherpa   F            1
3   Sherpa   M          115

In [14]:
# Define altitude bins (in meters) based on key landmarks along the standard Everest climbing route.
# These values are chosen to reflect the approximate elevations of each major camp and the summit region.

altitude_bins = [0, 5360, 6035, 6474, 7158, 7906, 8900]
altitude_labels = [
   'Below Base Camp', # Includes all deaths that occurred below Everest Base Camp (5360 m)
   'Base Camp Area', # Base Camp is at ~5360 m; this bin covers the area from BC up to Camp I (~6035 m)
   'Camp I Area', # Camp I is around 6035 m; this bin spans from Camp I to Camp II (~6474 m)
   'Camp II Area', # Camp II is around 6474 m; this bin spans up to Camp III (~7158 m)
   'Camp III Area', # Camp III is around 7158 m; this bin spans up to Camp IV (~7906 m)
   'Camp IV-Summit Area' # Camp IV is around 7906 m; this bin includes deaths from Camp IV to the summit (~8849 m)
]

    

# Create the altitude groups and count deaths
death_height = (
   death_data
   .assign(
       altitude_group=lambda x: pd.cut(x['deathhgtm'], 
                                     bins=altitude_bins,
                                     labels=altitude_labels,
                                     include_lowest=True)
   )
   .dropna(subset=['deathhgtm'])
   .groupby(['group', 'classification', 'altitude_group'])
   .size()
   .reset_index(name='death_count')
   .query('death_count > 0')
)

# Reorder altitude_group from TOP to BOTTOM
altitude_order = [
   'Camp IV-Summit Area',
   'Camp III Area', 
   'Camp II Area',
   'Camp I Area', 
   'Base Camp Area', 
   'Below Base Camp'
]

death_height['altitude_group'] = pd.Categorical(
   death_height['altitude_group'], 
   categories=altitude_order, 
   ordered=True
)

death_height = death_height.sort_values(['altitude_group', 'group'])

# Save to CSV

death_height.to_csv("death_height_sherpa_climbers.csv", index=False)


/var/folders/j7/bs_78v014_z3vp4jh1n05bth0000gn/T/ipykernel_13307/3269874230.py:26: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  .groupby(['group', 'classification', 'altitude_group'])


In [16]:

# Filtering gender to keep only 'M' and 'F'

everest_death_by_decade = everest[everest['sex'].isin(['M', 'F'])].copy()

# Creating 'decade' column

everest_death_by_decade['decade'] = (everest_death_by_decade['myear'] // 10) * 10
everest_death_by_decade['decade'] = everest_death_by_decade['decade'].astype(int).astype(str) + 's'

# Grouping by decade and gender for Everest climb attempts and expedition deaths

death_stats = (
    everest_death_by_decade
    .groupby(['decade', 'sex'])
    .agg(
        attempts=('membid', 'count'),
        deaths=('death', 'sum')
    )
    .reset_index()
)

# Calculating death rate per attempt (%)

death_stats['Death Rate per Attempt (%)'] = round(
    (death_stats['deaths'] / death_stats['attempts']) * 100, 2
)

# Renaming columns
death_stats = death_stats.rename(columns={
    'decade': 'Decade',
    'sex': 'Gender',
    'attempts': 'Attempts',
    'deaths': 'Deaths'
})


# Pivoting to wide format

death_stats_wide = death_stats.pivot(index='Decade', columns='Gender')

#Flatten MultiIndex columns

death_stats_wide.columns = [' '.join(col).strip() for col in death_stats_wide.columns.values]

#Renaming columns 

death_stats_wide = death_stats_wide.rename(columns={
    'Attempts M': 'Male Attempts',
    'Attempts F': 'Female Attempts',
    'Deaths M': 'Male Deaths',
    'Deaths F': 'Female Deaths',
    'Death Rate per Attempt (%) M': 'Male Death Rate (%)',
    'Death Rate per Attempt (%) F': 'Female Death Rate (%)'
})

# Reset index to convert 'Decade' from index to column
death_stats_wide = death_stats_wide.reset_index()

#Fill missing values with 0
death_stats_wide = death_stats_wide.fillna(0)

print(death_stats_wide)

#Save
death_stats_wide.to_csv("death_by_gender.csv", index=False)


   Decade  Female Attempts  Male Attempts  Female Deaths  Male Deaths  \
0   1920s              0.0           96.0            0.0         26.0   
1   1930s              0.0          121.0            0.0          1.0   
2   1940s              0.0            3.0            0.0          0.0   
3   1950s              1.0          108.0            0.0          1.0   
4   1960s              0.0          167.0            0.0          6.0   
5   1970s             43.0          599.0            1.0         28.0   
6   1980s            122.0         1944.0            1.0         59.0   
7   1990s            284.0         3221.0            3.0         59.0   
8   2000s            510.0         5840.0            2.0         49.0   
9   2010s            803.0         8053.0            6.0         77.0   
10  2020s            381.0         3063.0            1.0         33.0   

    Female Death Rate (%)  Male Death Rate (%)  
0                    0.00                27.08  
1                    0.00

In [19]:
# Total attempts and deaths by decade with death rate

total_stats_by_decade = (
   everest_death_by_decade
   .groupby('decade')
   .agg(
       total_attempts=('membid', 'count'),
       total_deaths=('death', 'sum')
   )
   .reset_index()
)

# Calculate death rate per attempt (%)

total_stats_by_decade['Death Rate per Attempt (%)'] = round(
   (total_stats_by_decade['total_deaths'] / total_stats_by_decade['total_attempts']) * 100, 2
)

# Rename columns
total_stats_by_decade = total_stats_by_decade.rename(columns={
   'decade': 'Decade',
   'total_attempts': 'Total Attempts',
   'total_deaths': 'Total Deaths'
})

print(total_stats_by_decade)


   Decade  Total Attempts  Total Deaths  Death Rate per Attempt (%)
0   1920s              96            26                       27.08
1   1930s             121             1                        0.83
2   1940s               3             0                        0.00
3   1950s             109             1                        0.92
4   1960s             167             6                        3.59
5   1970s             642            29                        4.52
6   1980s            2066            60                        2.90
7   1990s            3505            62                        1.77
8   2000s            6350            51                        0.80
9   2010s            8856            83                        0.94
10  2020s            3444            34                        0.99


In [24]:
# Filter out invalid age values and create age groups

age_death_analysis = everest[
   (everest['calcage'] > 0) & 
   (everest['calcage'].notna())
].copy()

# Create 5-year age groups

age_death_analysis['age_group'] = (age_death_analysis['calcage'] // 5) * 5
age_death_analysis['age_group'] = age_death_analysis['age_group'].astype(int).astype(str) + '-' + (age_death_analysis['age_group'] + 4).astype(int).astype(str)

# Group by age group for attempts and deaths

age_group_stats = (
   age_death_analysis
   .groupby('age_group')
   .agg(
       total_attempts=('membid', 'count'),
       total_deaths=('death', 'sum')
   )
   .reset_index()
)

# Calculate death rate per attempt (%)

age_group_stats['Death Rate per Attempt (%)'] = round(
   (age_group_stats['total_deaths'] / age_group_stats['total_attempts']) * 100, 2
)

# Rename columns
age_group_stats = age_group_stats.rename(columns={
   'age_group': 'Age Group',
   'total_attempts': 'Total Attempts',
   'total_deaths': 'Total Deaths'
})

print(age_group_stats)

#age_group_stats.to_csv("death_rate_by_age_group.csv", index=False)

   Age Group  Total Attempts  Total Deaths  Death Rate per Attempt (%)
0      10-14               4             0                        0.00
1      15-19             298             2                        0.67
2      20-24            1942            23                        1.18
3      25-29            4113            48                        1.17
4      30-34            4903            59                        1.20
5      35-39            4392            51                        1.16
6      40-44            3502            41                        1.17
7      45-49            2439            28                        1.15
8      50-54            1520            24                        1.58
9      55-59             826            10                        1.21
10     60-64             400            11                        2.75
11     65-69             151             3                        1.99
12     70-74              44             0                        0.00
13    

In [80]:
sherpa_women = everest[
   everest['death'] & 
   (everest['sex'] == 'F') & 
   (everest['sherpa'] == True)
][['expid','fname', 'lname', 'sex', 'citizen', 'calcage', 'status', 'msolo', 'deathhgtm', 'deathdate', 'deathclass','deathnote']]

print(sherpa_women)

          expid         fname   lname sex citizen  calcage  status  msolo  \
3286  EVER93111  Pasang Lhamu  Sherpa   F   Nepal       30  Leader  False   

      deathhgtm  deathdate  deathclass  \
3286       8750 1993-04-23           5   

                                              deathnote  
3286  Died of exhaustion and exposure early morning ...  


In [59]:
# 5 = Death during the descent phase of a summit bid
# The 'deathclass' column contains numerical codes representing the phase of death, which are taken directly from the Himalayan Database Guide, 
# available at: https://www.himalayandatabase.com/downloads/Himalayan%20Database%20Guide.pdf

### The only woman who died on Mt. Everest from the Sherpa community is Pasanga Lhamu Sherpa. She was the first Nepali woman to summit Everest.
### She died during the descent phase of a summit on April 23, 1993, at an altitude of 8750 meters, just below the summit. 
### She was a leader of the 1993 government-sponsored Nepali Women's Everest Expedition.